pkg/controller/pxc/vs.go

```diff
@@ -1,72 +1,77 @@
 package pxc
 
 import (
-	"encoding/json"
 	"fmt"
-	"net/http"
+	"net/url"
 	"time"
+
+	"github.com/percona/percona-xtradb-cluster-operator/versionserviceclient"
+	"github.com/percona/percona-xtradb-cluster-operator/versionserviceclient/models"
+	"github.com/percona/percona-xtradb-cluster-operator/versionserviceclient/version_service"
 )
 
-func (vs VersionServiceClient) GetExactVersion(desiredVersion string, currentVersion string) (DepVersion, error) {
-	client := http.Client{
-		Timeout: 5 * time.Second,
-	}
-	url := fmt.Sprintf("%s/api/versions/v1/pxc/%s/%s?databaseVersion=%s", vs.URL, vs.OpVersion, desiredVersion, currentVersion)
-	req, err := http.NewRequest("GET", url, nil)
-	if err != nil {
-		return DepVersion{}, err
-	}
-	req.Header.Set("Accept", "application/json")
+const productName = "pxc"
 
-	resp, err := client.Do(req)
+func (vs VersionServiceClient) GetExactVersion(desiredVersion, currentVersion string, versionMeta currVersionMeta) (DepVersion, error) {
+	requestURL, err := url.Parse(vs.URL)
 	if err != nil {
 		return DepVersion{}, err
 	}
 
-	defer resp.Body.Close()
-	if resp.StatusCode != http.StatusOK {
-		return DepVersion{}, fmt.Errorf("received bad status code %s", resp.Status)
-	}
+	srvCl := versionserviceclient.NewHTTPClientWithConfig(nil, &versionserviceclient.TransportConfig{
+		Host:     requestURL.Host,
+		BasePath: requestURL.Path,
+		Schemes:  []string{requestURL.Scheme},
+	})
+
+	params := version_service.NewVersionServiceApplyParamsWithTimeout(10 * time.Second).WithProduct(productName).
+		WithOperatorVersion(vs.OpVersion).WithApply(desiredVersion).WithDatabaseVersion(&currentVersion).
+		WithKubeVersion(&versionMeta.KubeVersion).WithPlatform(&versionMeta.Platform).WithCustomResourceOid(&versionMeta.CRUID).
+		WithPmmVersion(&versionMeta.PMMVersion).WithBackupVersion(&versionMeta.BackupVersion)
```

Comment:
I think you don't have to use all those methods to set struct field

Reply:
Yea, right. Ty